pkg/osquery/tables/firefox_preferences/table.go

```diff
@@ -0,0 +1,105 @@
+package firefox_preferences
+
+import (
+	"bufio"
+	"context"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/launcher/pkg/dataflatten"
+	"github.com/kolide/launcher/pkg/osquery/tables/dataflattentable"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	"github.com/osquery/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+type Table struct {
+	name   string
+	logger log.Logger
+}
+
+const tableName = "kolide_firefox_preferences"
+
+var re = regexp.MustCompile(`^user_pref\((.*)\)`)
+
+func TablePlugin(logger log.Logger) *table.Plugin {
+	columns := dataflattentable.Columns(
+		table.TextColumn("path"),
+	)
+
+	t := &Table{
+		name:   tableName,
+		logger: logger,
+	}
+
+	return table.NewPlugin(t.name, columns, t.generate)
+}
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	return generateData(queryContext, t.logger)
+}
+
+func generateData(queryContext table.QueryContext, logger log.Logger) ([]map[string]string, error) {
+	var results []map[string]string
```

Comment:
can we move this down to closer to where it's used? just before we start the for loops below

Reply:
FWIW we often stick results at the top, since all the loops feed into it. I don't have strong feelings.

Reply:
I'm cool with moving it, but I was copying a pattern from `kolide_cryptoinfo`.

Reply:
since we never error, it may be needed sooner, so it's good as it is =)